src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-max-event-pool-size.mdx

```diff
@@ -0,0 +1,565 @@
+---
+
+<Tabs>
+	<TabsBar>
+        <TabsBarItem id="android">
+            Android
+        </TabsBarItem>
+        <TabsBarItem id="ios">
+            iOS
+        </TabsBarItem>
+        <TabsBarItem id="capacitor">
+            Capacitor
+        </TabsBarItem>
+        <TabsBarItem id="cordova">
+            Cordova
+        </TabsBarItem>
+        <TabsBarItem id="flutter">
+            Flutter
+        </TabsBarItem>
+        <TabsBarItem id="react">
+            React Native
+        </TabsBarItem>
+        <TabsBarItem id="xamarin">
+            Xamarin
+        </TabsBarItem>
+        <TabsBarItem id="maui">
+            .NET MAUI
+        </TabsBarItem>
+    </TabsBar>
+
+    <TabsPages>
+        <TabsPageItem id="android">
+## Syntax [#syntax]
+
+### Java [#java]
+
+```java
+NewRelic.setMaxEventPoolSize(int $maxSize)
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
+
+## Description [#description]
+
+Sets the maximum size of the event pool.
+
+By default, mobile monitoring collects a maximum of 1,000 events per event harvest cycle, which is 600 seconds long by default.
+This method controls the maximum size of the event pool stored in the memory until the next harvest cycle. When the pool size limit is reached, the New Relic Android agent will begin  [sampling events](/docs/agents/manage-apm-agents/agent-data/new-relic-events-limits-sampling), discarding some old and some new events, until the pool of events are transmitted with the next harvest cycle.
+This method lets you override the maximum size of that event pool. 
+
+* The default value for the event harvest cycle is 600 seconds.
+
+See also [`setMaxEventBufferTime()`](/docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/setmaxeventbuffertime-android-sdk-api), which lets you change the length of the event harvest cycle.
+
+<Callout variant="important">
+  Be aware that reporting a large number of events, or reporting events too frequently, may impact app performance.
+</Callout>
+
+For context on how to use this API, see [Send custom attributes and events](/docs/mobile-monitoring/new-relic-mobile-android/install-configure/work-android-sdk-api#attributes-events-insights).
+
+## Parameters [#parameters]
+
+<table>
+  <thead>
+    <tr>
+      <th width={200}>
+        Parameter
+      </th>
+      <th width={200}>
+        Type
+      </th>
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$maxSize`
+      </td>
+
+            <td>
+        `int`
+      </td>
+
+      <td>
+        Required. Maximum size of event pool.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Return values [#return-values]
+
+Returns `true` if it succeeds, or `false` if it doesn't.
+
+## Example [#example]
+
+Here's an example to set max size of event pool to 1000:
+
+### Java [#java]
+
+```java
+boolean poolSizeSet = NewRelic.setMaxEventPoolSize(1000);
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
```

Comment:
@ndesai-newrelic Can you please provide a kotlin example equivalent to the Java example?

Reply:
need to discuss with you

Reply:
```suggestion
```kotlin
    NewRelic.setMaxEventPoolSize(1000)
```
```